src/emulator/controller.ts

```diff
@@ -60,8 +60,13 @@ export async function startEmulator(instance: EmulatorInstance): Promise<void> {
   if (!portOpen) {
     await cleanShutdown();
     const description = Constants.description(name);
-    utils.logWarning(`Port ${port} is not open on ${host}, could not start ${description}.`);
-    utils.logBullet(`To select a different host/port for the emulator, update your "firebase.json":
+    utils.logLabeledWarning(
+      name,
+      `Port ${port} is not open on ${host}, could not start ${description}.`
+    );
+    utils.logLabeledBullet(
+      name,
+      `To select a different host/port for the emulator, update your "firebase.json":
```

Comment:
Will this warning get thrown if they haven't even passed in a firebase.json?  Or is it thrown _only_ when something specified in their firebase.json is wrong?

Reply:
This one is thrown any time we try to bind the emulator to port X and port X is already busy.

Reply:
```suggestion
      `To select a different host/port for the emulator, specify that host/port in a firebase.json config file:
```